include/soci/soci-platform.h

```diff
@@ -124,6 +124,119 @@ namespace cxx_details
     using std::auto_ptr;
 #endif
 
+#if defined(SOCI_HAVE_CXX11) || (defined(_MSC_VER) && _MSC_VER >= 1800)
+    #include <memory>
+    template <typename T>
+    using shared_ptr = std::shared_ptr<T>;
+
+    template<class T, class U>
+    inline shared_ptr<T> static_pointer_cast(shared_ptr<U> const& r)
+    { return std::static_pointer_cast<T>(r); }
+
+    template<class T, class U>
+    inline shared_ptr<T> dynamic_pointer_cast(shared_ptr<U> const& r)
+    { return std::dynamic_pointer_cast<T>(r); }
+
+    template<class T, class U>
+    inline shared_ptr<T> const_pointer_cast(shared_ptr<U> const& r)
+    { return std::const_pointer_cast<T>(r); }
+
+#else // std::shared_ptr<> not available
+    template <typename T>
+    class shared_ptr
```

Comment:
If we do decide to use our own shared pointer class, we could make it intrusive, which would be both more efficient and a bit simpler.

Reply:
I am not sure if I understand what you mean by "intrusive" here.

Anyways, I would like to highlight just two points before we go deeper: 

1.    currently I am using this class on 2 different points where they are really needed (could not use `unique_ptr` or `auto_ptr`). 
2.    If we solve the confusion between `class blob;` and `class blob_backend;` as reported on the issue 922, I will likely not need this class anymore.